api/handlers_api.go

```diff
@@ -110,42 +89,76 @@ func (r *Request) HandleJoinPlayer() (*mc.Message[mc.RespJoinGame], *mb.Game) {
 	r.Session.Player = joinPlayer
 
 	resp.AddPayload(mc.RespJoinGame{GameUuid: game.Uuid, PlayerUuid: joinPlayer.Uuid})
-	return &resp, game
+	return resp, game
+}
+
+// User will choose the configurations of ships on defence grid.
+// Then the grid is sent to backend and adjustment happens accordingly.
+func (r *Request) HandleReadyPlayer() (mc.Message[mc.NoPayload], *mb.Game) {
+	var readyPlayerReq mc.Message[mc.ReqReadyPlayer]
+	resp := mc.NewMessage[mc.NoPayload](mc.CodeReady)
+
+	if err := json.Unmarshal(r.Payload, &readyPlayerReq); err != nil {
+		resp.AddError(err.Error(), cerr.ConstErrInvalidPayload)
+		return resp, nil
+	}
+
+	game, player, err := r.Session.GameManager.FindGameAndPlayer(readyPlayerReq.Payload.GameUuid, readyPlayerReq.Payload.PlayerUuid)
```

Comment:
since we are only using `GridSize` from the game, I'd suggest against returning the whole game object here. You can have two separate functions: 
1. `player = FindPlayer()`
2. `gridSize = FindGridSize()`

Reply:
Modified finding the player logic.

I see no point having another function for grid size since we're returning the game anyways and can access the grid size easily.